adapters/adapters-base/src/main/java/org/apache/cassandra/sidecar/adapters/base/CassandraCompactionManagerOperations.java

```diff
@@ -53,4 +55,31 @@ public List<Map<String, String>> getCompactions()
         return jmxClient.proxy(CompactionManagerJmxOperations.class, COMPACTION_MANAGER_OBJ_NAME)
                         .getCompactions();
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void stopCompactionById(String compactionId)
+    {
+        // compactionId takes precedence over type if both are provided
+        if (compactionId != null && !compactionId.trim().isEmpty()) {
+            CompactionManagerJmxOperations proxy = jmxClient.proxy(
+                    CompactionManagerJmxOperations.class,
+                    COMPACTION_MANAGER_OBJ_NAME
+            );
+            proxy.stopCompactionById(compactionId);
+        }
```

Comment:
Do you want to throw an exception when the compactionId is invalid here and in the `stopCompaction` method